src/components/Reports/ReportsPageRanking/ReportsPageRanking.js

```diff
@@ -11,97 +11,141 @@ const numberFormatOptions = {
   maximumFractionDigits: 2,
 };
 
+const pageSize = 2;
+const initialState = { pages: [], page: 0, loading: true };
+
 const ReportsPageRanking = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
-  const [state, setState] = useState({ loading: true });
+  const [state, dispatchListEvent] = useReducer((prevState, action) => {
+    switch (action.type) {
+      case 'loadingMore':
+        return { ...prevState, loading: true };
+      case 'moreLoaded':
+        return { pages: [...prevState.pages, ...action.pages], page: prevState.page + 1 };
+      case 'errorOnLoad':
+        return { ...prevState, loading: false, error: true };
+      case 'reset':
+      default:
+        return initialState;
+    }
+  }, initialState);
+
+  const fetchData = async (datahubClient, domainName, dateFrom, pageNumber) => {
+    dispatchListEvent({ type: 'loadingMore' });
+    const result = await datahubClient.getPagesRankingByPeriod({
+      domainName: domainName,
+      dateFrom: dateFrom,
+      pageSize: pageSize,
+      pageNumber: pageNumber,
+    });
+    if (!result.success) {
+      dispatchListEvent({ type: 'errorOnLoad' });
+    } else {
+      dispatchListEvent({ type: 'moreLoaded', pages: result.value });
+    }
+  };
+
+  const showMoreResults = () => {
+    fetchData(datahubClient, domainName, dateFrom, state.page + 1);
```

Comment:
```suggestion
    return fetchData(datahubClient, domainName, dateFrom, state.page + 1);
```

Or simply remove the `{`:

```js
const showMoreResults = () =>
    fetchData(datahubClient, domainName, dateFrom, state.page + 1);
```